### ISSUE TITLE
latitude longitude in aimsicd-locationinfo.csv incorrect

### ISSUE THREAD

**Issue by indoraq** [SecUpwN/Android-IMSI-Catcher-Detector#327]
latitude and longitude valuse written to aimsicd-locationinfo.csv on DB backup change only when CellID has changed and thus provide a very bad trajectory:

"_id"	"Lac"	"CellID"	"Net"	"Lat"	"Lng"	"Timestamp"
"11712"	"1300"	"6801452"	"10"	"58.37453215"	"26.73167696"	"2015-02-16 07:58:32"
"11713"	"1300"	"6801452"	"10"	"58.37453215"	"26.73167696"	"2015-02-16 07:58:44"
"11714"	"1300"	"6801452"	"10"	"58.37453215"	"26.73167696"	"2015-02-16 07:58:52"
"11715"	"1300"	"6801452"	"10"	"58.37453215"	"26.73167696"	"2015-02-16 07:59:01"
"11716"	"1300"	"6801452"	"10"	"58.37453215"	"26.73167696"	"2015-02-16 07:59:12"
"11717"	"1300"	"6801452"	"10"	"58.37453215"	"26.73167696"	"2015-02-16 07:59:18"
"11718"	"1300"	"6801452"	"10"	"58.37453215"	"26.73167696"	"2015-02-16 07:59:28"
"11719"	"1300"	"6801452"	"10"	"58.37453215"	"26.73167696"	"2015-02-16 07:59:38"
"11720"	"1300"	"6801452"	"10"	"58.37453215"	"26.73167696"	"2015-02-16 07:59:51"
"11721"	"1300"	"6807963"	"10"	"58.37535558"	"26.73032492"	"2015-02-16 08:00:00"
"11722"	"1300"	"6807963"	"10"	"58.37535558"	"26.73032492"	"2015-02-16 08:00:11"
"11723"	"1300"	"6807963"	"10"	"58.37535558"	"26.73032492"	"2015-02-16 08:00:22"
"11724"	"1300"	"6807963"	"10"	"58.37535558"	"26.73032492"	"2015-02-16 08:00:29"
"11725"	"1300"	"6807963"	"10"	"58.37535558"	"26.73032492"	"2015-02-16 08:00:40"
"11726"	"1300"	"6807963"	"10"	"58.37535558"	"26.73032492"	"2015-02-16 08:00:50"
"11727"	"1300"	"6807963"	"10"	"58.37535558"	"26.73032492"	"2015-02-16 08:01:00"
"11728"	"1300"	"6806666"	"10"	"58.37142229"	"26.71736041"	"2015-02-16 09:05:16"

**Comment by E3V3A** [comment]
Oouch! But it could be on purpose in order to conserve battery when not moving around enough. We put the GPS to sleep when not moving around... More testing is required here in order to determine if this is a *bug* or a *feature*... Thanks for reporting!

**Comment by indoraq** [comment]
ok, then I suggest two modes of operation:
1. GPS-fix only on cell change (to conserve battery) Then there is also no need to bloat the measurement database with timestamps.
2. continuous GPS-fix (or tracking), where GPS fixes are obtained


As currently (at least what I have understood) AIMSICD can contribute to OpenCellID measurement data, then however GPS-fixes only on CGI change would lead to a biased data.

Note also that for a scenario when GPS-fix can not be obtained, there will be no location data currently recorded. This is incorrect, as CGI-s are there and can change even a lot. Should I submit for this a separate issue?

**Comment by E3V3A** [comment]
The thing is we need to remake the measurement table. (See #215 )  The issue we're having is that the device is polling info and not using listeners to save the network changes. Thus we're loosing many rapid network changes. But I don't know how to fix this right now. Once this is done, we can make sure to only save non-GPS related changes to the measurement table.

> Should I submit for this a separate issue?
Yes, please do and try explain more clearly. Thanks again.
